src/main/java/frc/robot/subsystems/IndexerSubsystem.java

```diff
@@ -1,126 +1,133 @@
   }
-
-  public boolean isShooterSensorTripped() {
-    return colorSensorReader.getShooterValues().proximity > THRESHOLD_SHOOTER;
+  
+  public boolean shouldContinue() {
+    return indexerColorMatch.matchClosestColor(indexerColorSensor.getColor()).color == COLOR_NOTE;
   }
 
-  public void stop() {
-    pidController.setReference(0, ControlType.kVelocity);
+  // Check actual motor values for dash board debugging
+  public boolean isActive() {
+    return leftIndexerMotor.get() > 0 || rightIndexerMotor.get() > 0;
   }
 
-  public void load() {
-    pidController.setReference(IndexerConstants.BELT_RUN_SPEED, ControlType.kVelocity);
+  public void runIndexer() {
+    leftPidController.setReference(BELT_RUN_SPEED, ControlType.kVelocity);
+    rightPidController.setReference(BELT_RUN_SPEED, ControlType.kVelocity);
   }
-  private final CANSparkMax leftIndexerMotor = new CANSparkMax(DEVICE_ID_INDEXER_LEFT_MOTOR, kBrushless);
-  private final CANSparkMax rightIndexerMotor = new CANSparkMax(DEVICE_ID_INDEXER_RIGHT_MOTOR, kBrushless);
-  private static SparkPIDController pidController;
-  private final ColorSensorReader colorSensorReader = new ColorSensorReader();
-  private final Notifier colorSensorNotifier = new Notifier(colorSensorReader);
-
-  private SysIdRoutine indexerSysIdRoutine = new SysIdRoutine(
-    new SysIdRoutine.Config(null, null, null, SysIdRoutineSignalLogger.logState()),
-    new SysIdRoutine.Mechanism((volts) -> {
-      leftIndexerMotor.setVoltage(volts.in(Volts));
-      rightIndexerMotor.setVoltage(volts.in(Volts));
-    }, null, this));
-
-  public IndexerSubsystem() {
-    indexerMotorConfig(leftIndexerMotor, true);
-    indexerMotorConfig(rightIndexerMotor, false);
 
-    colorSensorReader.run();
-    // Update the color sensors in the background to prevent loop overrun
-    colorSensorNotifier.setName("Color Sensors");
-    colorSensorNotifier.startPeriodic(0.02);
+  public void stopIndexer() {
+    leftPidController.setReference(0, ControlType.kVelocity);
+    rightPidController.setReference(0, ControlType.kVelocity);
+  } 
 
+  // Intake is intended to be called multiple times as it's called in "execute" for "IntakeCommand.java"
+  public void intake() {
+    if (shouldContinue()) {
+      runIndexer();
+    } else {
+      stopIndexer();
+    }
   }
 
-  public void addDashboardWidgets(ShuffleboardLayout dashboard) {
-    var detailDashboard = dashboard.getLayout("Detail", BuiltInLayouts.kGrid)
-        .withProperties(Map.of("Number of columns", 2, "Number of rows", 3)).withPosition(0, 0);
-    detailDashboard.addNumber("Intake Prox", () -> colorSensorReader.getIntakeValues().proximity).withPosition(0, 0);
-    detailDashboard.addNumber("Shooter Prox", () -> colorSensorReader.getIntakeValues().proximity).withPosition(0, 0);
-    detailDashboard.addString("Ring Type", () -> this.getIndexerState().name()).withPosition(1, 1);
-
+  private SysIdRoutine indexerSysIdRoutine = new SysIdRoutine(
```

Comment:
Fields should go before the constructor. This will compile and work, but it's confusing to humans. The order should be:
static fields
fields
constructors
methods (usually public then private, but not as important)